### DISCUSSION TITLE
Memray claiming completion even though the script doesn't complete

### DISCUSSION THREAD

**Discussion by rhoens** [bloomberg/memray#621] [Q&A]
I have a reasonable size code base with a large number of imports of various kinds for my script. When running the script, `memray` will at times claim that the script is done and the capture successful, even though the run failed doing imports (validated by observing print statements and nothing in the dump is past the import stage). I don't have a great repro as I can't share the code, but just wondering if anyone else has experienced this or have any ideas to get more diagnostics out of `memray` that would indicate what is actually happening.

**Comment by godlygeek** [comment]
"Completion" to Memray just means that the tracked process has stopped running, not necessarily that it ran successfully. Are you getting an `ImportError`? Or maybe something is calling `os.exit()`, or raising a `SystemExit` exception?
